python/dgl/graphbolt/sampled_subgraph.py

```diff
@@ -266,8 +267,15 @@ def _relabel_two_arrays(lhs_array, rhs_array):
     return mapping[: lhs_array.numel()], mapping[lhs_array.numel() :]
 
 
-def _exclude_homo_edges(edges, edges_to_exclude, assume_num_node_within_int32):
+def _exclude_homo_edges(
+    edges: Tuple[torch.Tensor, torch.Tensor],
+    edges_to_exclude: Tuple[torch.Tensor, torch.Tensor],
+    assume_num_node_within_int32: bool,
+):
     """Return the indices of edges that are not in edges_to_exclude."""
+    if edges_to_exclude is None:
+        # No edges are excluded, so return all indices of edges.
+        return torch.arange(len(edges[0]))
```

Comment:
Creating a large array tensor may incur some overhead. My local fix skips `_exclude_homo_edges` into the `index` dictionary, and then in the later `_slice_subgraph` and `_slice_subgraph_node_pairs` functions also skips the slicing operator. The basic idea is to use `None` as a special value for identity mapping. Not sure if the fix will impact other parts. Just a suggestion.

Reply:
the overhead exists even for `Not None`. But we could remove such overhead for `None`.

Reply:
`index[etype]=None` looks hacky as `index` indicates it stores the index to be selected.

Reply:
But for the performance issue, I do agree that we need to take into consideration. Probably we should do the reverse: `excluded_index` instead of `index` as excluded edges are usually a small portion of original edges?

Reply:
For now, I handle `None` specially as you suggested.

Reply:
> index[etype]=None looks hacky as index indicates it stores the index to be selected.

Agree. So I skip `index[etype] = ...` if the etype does not exist.

> But for the performance issue, I do agree that we need to take into consideration. Probably we should do the reverse: excluded_index instead of index as excluded edges are usually a small portion of original edges?

Yes. That could be a better option, but I'm not sure how will this complicate other parts.